lbz/events/broker.py

```diff
@@ -2,42 +2,35 @@
 from typing import Callable, Dict, List
 
 from lbz.events.event import Event
+from lbz.handler_base import HandlerBase
 from lbz.misc import get_logger
 
 logger = get_logger(__name__)
 
 
-class EventBroker:
+class EventBroker(HandlerBase):
     def __init__(
         self,
         mapper: Dict[str, List[Callable]],
-        raw_event: dict,
+        event: dict,
         *,
+        context: object = None,
         type_key: str = "detail-type",
         data_key: str = "detail",
     ) -> None:
+        self.context = context
         self.mapper = mapper
-        self.event = Event(raw_event[data_key], event_type=raw_event[type_key])
+        self.event = Event(event[data_key], event_type=event[type_key])
 
     def handle(self) -> None:
-        self.pre_handle()
-
         for handler in self._get_handlers():
             try:
                 handler(deepcopy(self.event))
```

Comment:
I am pretty sure you should pass `self.context` to each `handler`, otherwise, adding it to this class is pointless 😉

Reply:
In our current use case it will be mostly used in pre_handler